shared/teams/team/settings-tab/channel-popup.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as Container from '../../../util/container'
+import {pluralize} from '../../../util/string'
+
+type Props = {
+  disabledChannels?: Array<string>
+  onCancel: () => void
+  onComplete: (channels: Array<string>) => void
+  teamID: Types.TeamID
+}
+
+const ChannelPopup = (props: Props) => {
+  const {disabledChannels, onCancel, onComplete, teamID} = props
+  const [filter, setFilter] = React.useState('')
+  const filterLCase = filter.toLowerCase()
+  const onChangeFilter = (value: string) => setFilter(value)
+
+  const channelInfos = Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  const channels = [...channelInfos.values()].map(ci => ci.channelname)
+  const channelsFiltered = filter ? channels.filter(c => c.toLowerCase().includes(filterLCase)) : channels
+
+  const [selected, setSelected] = React.useState(new Set<string>())
+  const onSelect = (channelname: string) => {
+    if (selected.has(channelname)) {
+      selected.delete(channelname)
+    } else {
+      selected.add(channelname)
+    }
+    setSelected(new Set(selected))
+  }
+
+  const onAdd = () => onComplete([...selected])
+  return (
+    <Kb.MobilePopup overlayStyle={Styles.globalStyles.fullHeight}>
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.header} gap="tiny">
+        <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.headerTop}>
+          <Kb.Text type="BodyBigLink" onClick={onCancel}>
+            Cancel
+          </Kb.Text>
+          <Kb.Text
+            type="BodyBigLink"
+            onClick={selected.size ? onAdd : undefined}
+            style={!selected.size && styles.addDisabled}
+          >
+            Add{!!selected.size && ` (${selected.size})`}
+          </Kb.Text>
+        </Kb.Box2>
+        <Kb.SearchFilter
+          placeholderText={`Search ${channels.length} ${pluralize('channel', channels.length)}`}
+          size="full-width"
+          onChange={onChangeFilter}
+          style={styles.searchFilter}
+          placeholderCentered={true}
+          icon="iconfont-search"
+        />
+      </Kb.Box2>
+      <Kb.BoxGrow>
+        <Kb.ScrollView style={Styles.globalStyles.fillAbsolute}>
+          {channelsFiltered.map(channelname => {
```

Comment:
does this want to be a virtualized list if there are hundreds of channels?

Reply:
Ok, I'll add that.

Reply:
Done